### PR TITLE
Remove vendored googletest and update googletest version

### PR DESCRIPTION
**Pull request by alphan**
This PR:
* Switches to the googletest github repo from the copy in `sw/vendor`
* Removes the vendored copy in `sw/vendor`
* Fixes some minor issues in unit tests
* Updates googletest to ~release-1.12.0~ release-1.11.0 (the latest release has an undocumented dependency which breaks our builds, can update after that issue is resolved, see the commit message for details)

@timothytrippel please let me know if this requires an update to the image.

### CODE REVIEW

**Comment by dmcardle** [comment]
Oh no, I think this is nearly the same as #13397!

**Comment by dmcardle** [comment]
Aha, now I see why you used googletest release-1.11.0. I saw this message in [one of the CI logs](https://dev.azure.com/lowrisc/4c50229e-f6cb-4f39-8522-585bb3841438/_apis/build/builds/84990/logs/43) attached to #13397.

```
com_google_googletest/BUILD.bazel:80:11: no such package '@com_googlesource_code_re2//': The repository '@com_googlesource_code_re2' could not be resolved: Repository '@com_googlesource_code_re2' is not defined and referenced by '@com_google_googletest//:gtest'
```

Happy to close my PR, you seem to have fixed a few issues I hadn't run into yet, like the parameterized tests names.

**Comment by alphan** [comment]
> Happy to close my PR, you seem to have fixed a few issues I hadn't run into yet, like the parameterized tests names.

Sounds good, yes, almost identical :) There were some minor things I was waiting to fix until we switched to a version that threw errors for these cases.

File: sw/device/lib/dif/dif_keymgr_unittest.cc [hunk 1]
```diff
@@ -112,39 +112,6 @@ INSTANTIATE_TEST_SUITE_P(
...
-
-  void SetUp() override {
-    if (AllParamsGood()) {
-      // Only test negative cases.
-      GTEST_SKIP();
-    }
-  }
-};
-
-INSTANTIATE_TEST_SUITE_P(
```
Comment (drewmacrae) [inline sw/device/lib/dif/dif_keymgr_unittest.cc:136]: Are you removing a set of tests here intentionally?

**Review by drewmacrae — APPROVED** [review]

**Review by moidx — APPROVED** [review]

**Comment by timothytrippel** [comment]
@alphan, yeah thats 

> This PR:
> 
> * Switches to the googletest github repo from the copy in `sw/vendor`
> * Removes the vendored copy in `sw/vendor`
> * Fixes some minor issues in unit tests
> * Updates googletest to ~release-1.12.0~ release-1.11.0 (the latest release has an undocumented dependency which breaks our builds, can update after that issue is resolved, see the commit message for details)
> 
> @timothytrippel please let me know if this requires an update to the image.

@alphan yes this will require an update to the airgapped image. I can regenerate it then merge this.

**Comment by alphan** [comment]
Thanks a lot @timothytrippel!

**Pull request merged**